crates/wasmprinter/src/operator.rs

```diff
@@ -279,6 +279,12 @@ macro_rules! define_visit {
             $self.memory_index($mem)?;
         }
     );
+    (payload $self:ident MemoryDiscard $mem:ident) => (
+        if $mem != 0 {
+            $self.push_str(" ");
+            $self.memory_index($mem)?;
+        }
+    );
```

Comment:
If you'd like I think it's ok to remove this special case since not printing `memory.grow 0` is mostly just a compat with MVP wasm at this point. I think instructionsl ike `table.grow` always print their table index as they're newer, so always printing the memory index with `memory.discard` sounds ok to me at least.

Reply:
I tried removing this `MemoryDiscard` case entirely but I get the following error:

```
error[E0599]: no method named `mem` found for mutable reference `&mut PrintOperator<'_, '_>` in the current scope
   --> crates/wasmprinter/src/operator.rs:851:5
    |
851 |     wasmparser::for_each_operator!(define_visit);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: there is a method with a similar name: `memarg`
    |
    = note: this error originates in the macro `wasmparser::for_each_operator` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Not sure how to fix that, but just let me know and I'll update it.

Reply:
Ah ok there's a possible fix for this but I'll handle it later, so this is fine as-is. Thanks for testing though!